Source/Tabs/DataProviders/ShoppingList.lua

```diff
@@ -0,0 +1,129 @@
+local SHOPPING_LIST_TABLE_LAYOUT = {
+  {
+    headerTemplate = "AuctionatorStringColumnHeaderTemplate",
+    headerParameters = { "minPrice" },
+    headerText = "Price",
+    cellTemplate = "AuctionatorPriceCellTemplate",
+    cellParameters = { "minPrice" },
+    width = 140
+  },
+  {
+    headerTemplate = "AuctionatorStringColumnHeaderTemplate",
+    headerParameters = { "name" },
+    headerText = "Name",
+    cellTemplate = "AuctionatorItemKeyCellTemplate"
+  },
+  {
+    headerTemplate = "AuctionatorStringColumnHeaderTemplate",
+    headerText = "Available",
+    headerParameters = { "totalQuantity" },
+    cellTemplate = "AuctionatorStringCellTemplate",
+    cellParameters = { "totalQuantity" },
+    width = 70
+  }
+}
+
+local ITEM_KEY_EVENT = "ITEM_KEY_ITEM_INFO_RECEIVED"
+
+ShoppingListDataProviderMixin = CreateFromMixins(DataProviderMixin)
+
+function ShoppingListDataProviderMixin:OnLoad()
+  Auctionator.Debug.Message("ShoppingListDataProviderMixin:OnLoad()")
+
+  self.pendingItemIds = {}
```

Comment:
The stored item ids are never reused

Reply:
The `pendingItemIds` are used to unregister the "ITEM_KEY_ITEM_INFO_RECEIVED" event once there are no more `pendingItemIds`